src/rgw/rgw_lua_utils.h

```diff
@@ -82,6 +83,14 @@ class lua_state_guard {
                   const DoutPrefixProvider* _dpp);
   ~lua_state_guard();
   lua_State* get() { return state; }
+  void reset_start_time() { start_time = ceph::real_clock::now(); }
+
+  std::size_t get_max_memory() const { return max_memory; }
+  std::size_t get_mem_in_use() const { return mem_in_use; }
+  std::chrono::milliseconds get_max_runtime() const { return max_runtime; }
+  int set_max_memory(std::size_t _max_memory);
```

Comment:
in our code base the custom is that if a function returns integer result code, it could be treated as a linux error code (meaning: "0" is sucess and negative numer according to: https://www.thegeekstuff.com/2010/10/linux-error-codes/). If we just need success/failure from the function, you should return a boolean.